client/cli/go/cmds/tests/root_test.go

```diff
@@ -0,0 +1,165 @@
+package cmds
+
+import (
+	"io"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/codegangsta/negroni"
+	"github.com/gorilla/mux"
+	"github.com/heketi/heketi/apps/glusterfs"
+	"github.com/heketi/heketi/client/api/go-client"
+	"github.com/heketi/heketi/client/cli/go/cmds"
+	"github.com/heketi/heketi/middleware"
+	"github.com/heketi/tests"
+)
+
+var (
+	diff                    int
+	HEKETI_CLI_TEST_VERSION           = "testing"
+	sout                    io.Writer = os.Stdout
+	serr                    io.Writer = os.Stderr
+	TEST_ADMIN_KEY                    = "adminkey"
+	db                      string
+	app                     *glusterfs.App
+	Ts                      *httptest.Server
+	Url                     string
+)
+
+func setupHeketiServer(app *glusterfs.App) *httptest.Server {
+	router := mux.NewRouter()
+	app.SetRoutes(router)
+	n := negroni.New()
+
+	jwtconfig := &middleware.JwtAuthConfig{}
+	jwtconfig.Admin.PrivateKey = TEST_ADMIN_KEY
+	jwtconfig.User.PrivateKey = "userkey"
+
+	// Setup middleware
+	n.Use(middleware.NewJwtAuth(jwtconfig))
+	n.UseFunc(app.Auth)
+	n.UseHandler(router)
+
+	// Create server
+	return httptest.NewServer(n)
+}
+
+func TestVersion(t *testing.T) {
+	c := cmds.NewHeketiCli(HEKETI_CLI_TEST_VERSION, sout, serr)
+	db := tests.Tempfile()
+	//	defer os.Remove(db)
+
+	// Create the app
+	app := glusterfs.NewTestApp(db)
+	//	defer app.Close()
+
+	// Setup the server
+	Ts := setupHeketiServer(app)
```

Comment:
The setup needs to be done and cleaned up at each test.  Cannot assume test order.